.travis.yml

```diff
@@ -5,25 +5,19 @@ services:
 - docker
 install:
 - pip install --no-cache-dir ruamel.yaml python-dateutil
-- mkdir -p bin
-- export PATH=$PWD/bin:$PATH
-- curl -ssL https://storage.googleapis.com/kubernetes-helm/helm-v2.6.1-linux-amd64.tar.gz | tar -xz -C bin --strip-components 1 linux-amd64/helm
-- chmod +x bin/helm
-- helm init --client-only
+- . ci/minikube.env
+- ./ci/install.sh
 script:
 - |
-  if [[ "$TRAVIS_PULL_REQUEST" == "false" ]]; then
+  if [[ "$TRAVIS_PULL_REQUEST" == "false" && ( "$TRAVIS_BRANCH" = master || "$TRAVIS_BRANCH" == v0.4.x ) ]]; then
     docker login -u ${DOCKER_USERNAME} -p "${DOCKER_PASSWORD}"
     export PUSH="--push --publish-chart"
   fi
 - "./build.py build --commit-range ${TRAVIS_COMMIT_RANGE} $PUSH"
-branches:
-  only:
-  - master
-  - v0.4.x
+- ./ci/test.sh
 before_install:
 - |
-  if [[ "$TRAVIS_PULL_REQUEST" == "false" ]]; then
+  if [[ "$TRAVIS_PULL_REQUEST" == "false" && ( "$TRAVIS_BRANCH" = master || "$TRAVIS_BRANCH" == v0.4.x ) ]]; then
```

Comment:
This TRAVIS_BRANCH logic is redundant with the `branches.only` logic above that was removed. Can we switch back to that? Or did it not work for some reason?

Reply:
I changed that to ensure travis would always run on your own branches when developing. Otherwise you have to either develop on your own `master` branch, have a separate commit to modify travis and remove it before opening a PR, or only have it tested after opening the PR.

If I've understood correctly the original aim was to only do `docker push` for the two branches which makes sense, but I think unconditionally running the tests is useful- I usually have travis setup on my own repos/forks to run on every push on any branch. I'm happy to revert it though.

Reply:
Good point. What we should probably do is move these bits to a specific `deploy` step, like we do [on repo2docker](https://github.com/jupyter/repo2docker/blob/master/.travis.yml#L14).

I think this is AOK as it is right now.